nightfall-deployer/contracts/Challenges.sol

```diff
@@ -34,8 +34,11 @@ contract Challenges is Stateful, Key_Registry, Config {
     bytes32 salt
   ) external {
     checkCommit(msg.data);
+    require(blockL2.blockNumberL2 == priorBlockL2.blockNumberL2+1 || (blockL2.blockNumberL2 == 0 && priorBlockL2.blockNumberL2 == 0),'previous block number is incorrect');
```

Comment:
Yes . It is redundant but my thought was to exit if the first check itself fail. I will do a check for the block zero case separately. Would it be better to write the checks inside `ChallengesUtil.sol `and `Utils.sol` instead of `Challenges.sol` ?